pkg/systemd/generate/common.go

```diff
@@ -30,18 +30,23 @@ func validateRestartPolicy(restart string) error {
 	return errors.Errorf("%s is not a valid restart policy", restart)
 }
 
-const headerTemplate = `# {{{{.ServiceName}}}}.service
-# autogenerated by Podman {{{{.PodmanVersion}}}}
+const serviceTemplate = `# {{{{.ServiceName}}}}.service
```

Comment:
I prefer to use a conditional, which is more template idiomatic:
```go
const headerTemplate = `# {{{{.ServiceName}}}}.service
{{{- if.GenerateHeader}}}
# autogenerated by Podman {{{{.PodmanVersion}}}}
{{{{- if .TimeStamp}}}}
# {{{{.TimeStamp}}}}
{{{{- end}}}}
{{{{- end}}}}

[Unit]
Description=Podman {{{{.ServiceName}}}}.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target
`
```

Reply:
Hmm nice one, will see how it works

Reply:
Works nicely, I admit didn't know about this (fairly new to go). However, I changed logic to GenerateNoHeader. Firstly, because the flag is called --no-header, and secondly because the default value will be false. Then, there is no need to add it to every structure and set explicitly true, as well as to tests. Hope it's ok for you.